src/react/projects/spark-core-react/src/SprkMasthead/components/SprkMastheadSelector/SprkMastheadSelector.js

```diff
@@ -0,0 +1,233 @@
+/* global window */
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import classNames from 'classnames';
+import { uniqueId } from 'lodash';
+import SprkIcon from '../../../SprkIcon/SprkIcon';
+
+class SprkMastheadSelector extends Component {
+  constructor(props) {
+    super();
+    this.state = {
+      isOpen: false,
+      triggerText: props.defaultTriggerText,
+      choiceItems: props.choices.items.map(item => ({ id: uniqueId(), ...item })),
+    };
+    this.openDropdown = this.openDropdown.bind(this);
+    this.closeDropdown = this.closeDropdown.bind(this);
+    this.updateTriggerText = this.updateTriggerText.bind(this);
+    this.closeOnEsc = this.closeOnEsc.bind(this);
+    this.closeOnClickOutside = this.closeOnClickOutside.bind(this);
+    this.myRef = React.createRef();
```

Comment:
I suggest we call this `dropdownRef` or something thats more descriptive than `myRef`.